File: connecting-wall/src/models/wall.ts
export interface Wall{
    wallID:string,
    wallName:string,
    dateCreated:string,
    dateUpdated:string,
    userID:string,
    // groupAConnection:GroupConnection,
    // groupBConnection:GroupConnection,
    // groupCConnection:GroupConnection,
    // groupDConnection:GroupConnection,
    groupConnections : GroupConnection[]
    // groupATerms:GroupTerm[],
    // groupBTerms:GroupTerm[],
    // groupCTerms:GroupTerm[],
    // groupDTerms:GroupTerm[]
}
export interface GroupConnection{
    connectionId:string,
    connectionName:string,
    connectionGroup : string,
    terms : Term
}
export interface Term{
    termID:string,
    termName:string,
    groupConnectionId : string
}

File: Backend/Registration/Registration/Models/Wall.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Registration.Models
{
    public class Wall
    {
        [Key]
        public Guid WallID { get; set; }
        [Required]
        public string WallName { get; set; }
        [Required]
        public DateTime DateCreated { get; set; }
        public DateTime? DateUpdated { get; set; }
        [ForeignKey("UserID"), Required]
        public string UserID { get; set; }
        public virtual ApplicationUser User { get; set; }
        //public GroupConnection GroupAConnection { get; set; }
        //public GroupConnection GroupBConnection { get; set; }
        //public GroupConnection GroupCConnection { get; set; }
        //public GroupConnection GroupDConnection { get; set; }
    }
}
